v2/sender.go

```diff
@@ -0,0 +1,122 @@
+package v2
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/messaging/azservicebus"
+)
+
+const (
+	msgTypeField = "type"
+)
+
+// MessageBody is a type to represent that an input message body can be of any type
+type MessageBody any
+
+// CustomSender is a wrapper around the ServiceBus sender that allows for users to introduce middleware to modify the ServiceBus message before it's sent
+type CustomSender interface {
+	SendMessage(ctx context.Context, mb MessageBody, options ...func(msg *azservicebus.Message) error) error
+}
```

Comment:
drop this interface, this can be defined user side if they want to.
we shouldn't provide interfaces